%YAML:1.0

camera:
  type: "PINHOLE"

  width: 640
  height: 480
  scaler: 1

  fx: 367.4252
  fy: 367.4862
  cx: 336.2885
  cy: 216.9875

  k1: -0.3107
  k2: 0.0773
  d1: -0.0008
  d2: 0.0002

Tbc: !!opencv-matrix
  rows: 3
  cols: 4
  dt: d
  data: [0, -0.325568,  0.945519, 0.2,
           -1,         0,         0,   0,
            0, -0.945519, -0.325568, 0.2] 


track:
  verbose: 1
  fm_threshold: 1.0
  max_cornor_num: 200
  min_cornor_gap: 15
  track_frequency: 10

initial:
  verbose: 1
  imu_cnt: 200
  imu_dtime: 2
  imu_accl_cov: 0.1
  imu_gyro_cov: 0.1

msckf:
  verbose: 1
  noise_accl: 0.025
  noise_gyro: 0.00025
  noise_accl_bias: 0.001
  noise_gyro_bias: 0.0001
  noise_observation: 0.01225

  large_dp: 0.5
  large_dv: 0.3

  ikf_iters: 10
  sliding_window_lens: 20
  distance_threshold: 0.5
  angle_threshold: 15 # in deg
  track_threshold: 0.5

sfm:
  verbose: 1
  finetune: 1
  max_iter_cnt: 20
  max_try_cnt: 10
  max_cond_number: 20000
  min_dist: 0.1
  max_dist: 100
  converge_threshold: 5.e-7
  min_disparity_angle: 5 # in deg
  min_disparity_distance: 0.1


# save trajectory
save_trajectory: 0
save_path: "/home/ubuntu/catkin_ws/src/learn_msckf/results/trajectory.txt"
